src/wallet/qi-hdwallet.ts

```diff
@@ -319,71 +315,83 @@ export class QiHDWallet extends AbstractHDWallet {
-		const addrNode = this._getHDNodeForAddress(address);
-		const privKey = addrNode.privateKey;
-		const digest = keccak_256(message);
-		const signature = schnorr.sign(digest, getBytes(privKey));
-		return hexlify(signature);
-	}
-	
-	public async serialize(): Promise<SerializedQiHDWallet> {
-		const hdwalletSerialized = await super.serialize();
-		return {
-			outpoints: this._outpoints,
-			changeAddresses: Array.from(this._changeAddresses.values()),
-			gapAddresses: this._gapAddresses,
-			gapChangeAddresses: this._gapChangeAddresses,
-			...hdwalletSerialized,
-		};
-	}
-
-	public static async deserialize(serialized: SerializedQiHDWallet): Promise<QiHDWallet> {
-		const wallet = await super.deserialize<QiHDWallet>(serialized) as QiHDWallet;
-		// import the change addresses
-		wallet.importSerializedAddresses(wallet._changeAddresses, serialized.changeAddresses);
-
-		// import the gap addresses, verifying they exist in the wallet
-		for (const gapAddressInfo of serialized.gapAddresses) {
-			const gapAddress = gapAddressInfo.address;
-			if (!wallet._addresses.has(gapAddress)) {
-				throw new Error(`Address ${gapAddress} not found in wallet`);
-			}
-			wallet._gapAddresses.push(gapAddressInfo);
-
-		}
-		// import the gap change addresses, verifying they exist in the wallet
-		for (const gapChangeAddressInfo of serialized.gapChangeAddresses) {
-			const gapChangeAddress = gapChangeAddressInfo.address;
-			if (!wallet._changeAddresses.has(gapChangeAddress)) {
-				throw new Error(`Address ${gapChangeAddress} not found in wallet`);
-			}
-			wallet._gapChangeAddresses.push(gapChangeAddressInfo);
-		}
-
-		// validate the outpoints and import them
-		for (const outpointInfo of serialized.outpoints) {
-			// check the zone is valid
-			wallet.validateZone(outpointInfo.zone);
-			// check the outpoint address is known to the wallet
-			if (!wallet._addresses.has(outpointInfo.address)) {
-				throw new Error(`Address ${outpointInfo.address} not found in wallet`);
-			}
-			const outpoint = outpointInfo.outpoint;
-			// TODO: implement a more robust check for Outpoint
-			// check the Outpoint fields are not empty
-			if (outpoint.Txhash == null || outpoint.Index == null || outpoint.Denomination == null) {
-				throw new Error(`Invalid Outpoint: ${JSON.stringify(outpoint)} `);
-			}
-			wallet._outpoints.push(outpointInfo);
-		}
-		return wallet;
-
-	}
+    public getGapChangeAddressesForZone(zone: Zone): NeuteredAddressInfo[] {
+        this.validateZone(zone);
+        const gapChangeAddresses = this._gapChangeAddresses.filter((addressInfo) => addressInfo.zone === zone);
+        return gapChangeAddresses;
+    }
+
+    public async signMessage(address: string, message: string | Uint8Array): Promise<string> {
+        const addrNode = this._getHDNodeForAddress(address);
+        const privKey = addrNode.privateKey;
+        const digest = keccak_256(message);
```

Comment:
Lets use the `keccak256` method exported from `crypto/index.js` everywhere in this file in place of `keccak_256` from noble